site/website/blog/2020-09-18-cdc-meets-stream-processing.md

```diff
@@ -0,0 +1,178 @@
+the code you would need to build an in-memory replica of your MySQL
+database table would be something like:
+
+```java
+StreamSource<ChangeRecord> source = MySqlCdcSources.mysql("source")
+        .setDatabaseAddress("127.0.0.1")
+        .setDatabasePort(3306)
+        .setDatabaseUser("debezium")
+        .setDatabasePassword("dbz")
+        .setClusterName("dbserver1")
+        .setDatabaseWhitelist("inventory")
+        .setTableWhitelist("inventory.customers")
+        .build();
+
+Pipeline pipeline = Pipeline.create();
+pipeline.readFrom(source)
+        .withoutTimestamps()
+        .peek()
+        .writeTo(CdcSinks.map("customers",
+                r -> r.key().toMap().get("id"),
+                r -> r.value().toObject(Customer.class).toString()));
+
+JobConfig cfg = new JobConfig().setName("mysql-monitor");
+Jet.bootstrappedInstance().newJob(pipeline, cfg);
+```
+
+One last service that Jet aims to add to Debezium’s value is performance
+benchmarks. We want to measure both the performance impact of enabling
+CDC on specific databases and the event throughput. This work is still
+ongoing, we still have to publish our exact testing methodology, but the
+results obtained so far are available [here](https://jet-start.sh/docs/next/design-docs/005-cdc-sources#performance).
+
+## Architecture
+
+I have stated above that when Debezium is integrated into Jet, the
+latter takes on the role of service-provider as far as fault tolerance
+and scalability are concerned.
+
+Jet doesn't delegate its cluster management and fault tolerance concerns
+to an outside system like ZooKeeper. It reuses the groundwork
+implemented for Hazelcast IMDG: cluster management and the IMap, and
+adds its own implementation of Chandy-Lamport distributed snapshots. If
+a cluster member fails, Jet will restart the job on the remaining
+members, restore the state of processing from the last snapshot, and
+then seamlessly continue from that point. For further details, consult
+our [documentation on the
+topic](https://jet-start.sh/docs/next/architecture/fault-tolerance).
+
+Extending this functionality umbrella to cover Debezium has been
+surprisingly simple. All that was really needed was to add Debezium’s
+source offset to Jet’s snapshots. This way, whenever Jet needs to
+execute a recovery, it can also pass the recovered offset to Debezium,
+which in turn is able to resume the data flow from that offset.
+
+One other thing we did and might be worth mentioning is that the Jet
+integration also makes use of Debezium’s [new record state
+extraction](https://debezium.io/documentation/reference/1.2/configuration/event-flattening.html)
+SMT (Simple Message Transformation), for the purpose of message
+structure simplification. With this transformation in effect, only the
+"after" structure of the Debezium event envelope is processed by Jet.
+However, whether this is a good idea or not, only time will tell. I
```

Comment:
I can already tell you it was an unfortunate decision. It makes processing delete events difficult. I left out a delete operation from my elastic/cdc blog because of this.

Reply:
As discussed on Slack, this is not really so for deletion. Delete events do have state/data/body and it's how the record looked before being deleted. Where you indeed only see the current image and not the previous one too is updates. 

We can revisit this, but separately from the Blog. For now maybe I can improve some wording, to make it more clear, because if you have been mislead, then others will be too.